src/psyclone/psyir/transformations/nemo_min_trans.py

```diff
@@ -0,0 +1,168 @@
+
+    `R=MIN(A,B,C,..) R=A; if B<R R=B; if C<R R=C; ...`
+
+    '''
+    def __init__(self):
+        super(NemoMinTrans, self).__init__()
+        self._operator_name = "MIN"
+        self._classes = (BinaryOperation, NaryOperation)
+        self._operators = (BinaryOperation.Operator.MIN,
+                           NaryOperation.Operator.MIN)
+
+    def apply(self, node, symbol_table, options=None):
+        '''Apply the MIN intrinsic conversion transformation to the specified
+        node. This node must be an MIN NaryOperation. The MIN
+        NaryOperation is converted to the following equivalent inline code:
+
+        R=MIN(A,B,C,..) => R=A; if B<R R=B; if C<R R=C; ...
+
+        In the PSyIR this is implemented as a transform from:
+
+        R= ... MIN(A,B,C...) ...
+
+        to:
+
+        res_min=A
+        tmp_min=B
+        IF (tmp_min<res_min) res_min=tmp_min
+        tmp_min=C
+        IF (tmp_min<res_min) res_min=tmp_min
+        ...
+        R= ... res_min ...
+
+        where A,B,C... could be arbitrarily complex expressions.
+
+        A symbol table is required as the NEMO api does not currently
+        contain a symbol table and one is required in order to create
+        temporary variables whose names do not clash with existing
+        code. This non-standard argument is also the reason why this
+        transformation is currently limited to the NEMO api.
+
+        This transformation requires the operation node to be a
+        descendent of an assignment and will raise an exception if
+        this is not the case.
+
+        :param node: a MIN Binary- or Nary-Operation node.
+        :type node: :py:class:`psyclone.psyGen.BinaryOperation` or \
+        :py:class:`psyclone.psyGen.NaryOperation`
+        :param symbol_table: the symbol table.
+        :type symbol_table: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param options: a dictionary with options for transformations.
+        :type options: dictionary of string:values or None
+
+        :returns: 2-tuple of new schedule and memento of transform.
+        :rtype: (:py:class:`psyclone.nemo.NemoInvokeSchedule`, \
+                 :py:class:`psyclone.undoredo.Memento`)
+
+        '''
+        self.validate(node, symbol_table)
+
+        schedule = node.root
+        memento = Memento(schedule, self, [node, symbol_table])
+
+        oper_parent = node.parent
+        assignment = node.ancestor(Assignment)
+
+        # Create a temporary result variable.
+        res_var = symbol_table.new_symbol_name("res_min")
+        symbol_table.add(DataSymbol(res_var, DataType.REAL))
+
+        # Replace operation with a temporary (res_var).
+        oper_parent.children[node.position] = Reference(res_var,
+                                                        parent=oper_parent)
+
+        # res_var=A
+        lhs = Reference(res_var)
+        new_assignment = Assignment.create(lhs, node.children[0])
```

Comment:
create() methods are designed to build PSyIR from the bottom up so only take arguments which help to build the required child structures. I agree that there is little support for adding in PSyIR nodes to an existing PSyIR tree and that a parent node for create() nodes could be useful in that context (although it would be good to create child and parent relationships at the same time) but that was not being looked at when developing the create() methods.